src/ConstructiveSolidGeometry/plotting/VolumePrimitives/VolumePrimitives.jl

```diff
@@ -1,4 +1,13 @@
-@recipe function f(p::AbstractVolumePrimitive)
+function sample(p::AbstractVolumePrimitive{T}, spacing::T)::Vector{CartesianPoint{T}} where {T}
+    fs = surfaces(p)
+    vs = Vector{CartesianPoint{T}}()
+    for s in fs
+        append!(vs, sample(s, spacing))
+    end
+    vs
+end
+
+@recipe function f(p::AbstractVolumePrimitive; n_samples = 100)
     fs = surfaces(p)
```

Comment:
Move this to line 23, so that we only determine the `surfaces` when we actually need them.
```julia
        if haskey(plotattributes, :seriestype) && plotattributes[:seriestype] == :samplesurface
            sample(p, extremum(p)/n_samples)
        else
            [surfaces(p)...]
        end
```